File: C2/PDOO/Practicas/DeepSpaceJava/src/deepspace/GameUniverseToUI.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author Profesor
 */
public class GameUniverseToUI {
    SpaceStationToUI currentStation;
    EnemyToUI currentEnemy;
    
    GameUniverseToUI(SpaceStation station,EnemyStarShip enemy) {
        currentStation = station.getUIversion();
        currentEnemy= enemy.getUIversion();
    }

    public SpaceStationToUI getCurrentStation() {
        return currentStation;
    }

    public EnemyToUI getCurrentEnemy() {
        return currentEnemy;
    }
    
    
}

File: C2/PDOO/Practicas/DeepSpaceRuby/lib/GameUniverseToUI.rb
#encoding:utf-8

module Deepspace

# 19.4.17 - Translation from Java
# @author Profesor
  
class GameUniverseToUI 
  
  attr_reader :currentStation, :currentEnemy
    
    def initialize (station,enemy) 
        @currentStation = station.getUIversion()
        @currentEnemy= enemy.getUIversion()
    end

    public
    
    def to_s
      out = "Estación actual -> #{@currentStation}\n"
      out +="Enemigo actual --> #{@currentEnemy}"
      return out
    end
    
    
end # class

end # module